packages/devextreme/js/ui/menu/ui.menu.js

```diff
@@ -271,6 +271,23 @@ class Menu extends MenuBase {
         this._initAdaptivity();
     }
 
+    _isTargetOutOfComponent(relatedTarget) {
+        const isInsideRootMenu = $(relatedTarget).closest(`.${DX_MENU_CLASS}`).length !== 0;
+        const isInsideContextMenu = $(relatedTarget).closest(`.${DX_CONTEXT_MENU_CLASS}`).length !== 0;
+
+        const isTargetOutOfComponent = !(isInsideRootMenu || isInsideContextMenu);
+
+        return isTargetOutOfComponent;
```

Comment:
We missed the scenario when disabled item is clicked. In this case `relatedTarget=null` and submenu will be badly closed.

I suggest you to add a check that related target is specified (like it's done for [overlay outside click](https://github.com/DevExpress/DevExtreme/blob/24_1/packages/devextreme/js/ui/overlay/ui.overlay.js#L295) for example). If relatedTarget=null we don't need to close a submenu because no element on the page is focused => a11y rule cannot be broken.

Reply:
You're leaving out a few scenarios: when the user of a component intentionally triggers a focusout (why shouldn't we close the menu in that case?) and, for example, when the user focuses the address bar or something outside the page (also a debatable case, e.g. our DD overlays close if the focus leaves the page, why shouldn't the Menu overlay close then? Here's an example: https://codepen.io/marker-dao/pen/YzMZdXd?editors=1010. Based on this, I decided to still close Menu when relatedTarget === null).

We need to come up with an additional solution for the case with disabled items. I suggest to look for a solution in DDEditors.

I need more time to research case with new conditions.

Reply:
We investigated all the cases in more detail and decided not to close the menu if !relatedTarget.